### ISSUE TITLE
[Craft 4] Two adjacent Neo block lists randomly merged their content/child blocks together

### ISSUE THREAD

**Issue by ericdrosas87** [spicywebau/craft-neo#951]
### Bug Description

![Screenshot 2024-09-30 at 14 14 38](https://github.com/user-attachments/assets/d7a22b8c-491f-42a8-8526-ee53d96a07e9)

Three days ago a content creator updated a page that contains Neo block fields and reported everything looking saving as expected. Today a different content creator went in to review the changes and noticed a message that stated `“changes have been merged into this version”` and then saw that the `Accordion Group` child blocks had merged with the `CTA Grid` blocks. There were no saves between three days ago and when the content creator discovered the merged child blocks. That is also to say that there are no revisions in the revision history that show where this change could have occurred. There were also no updates to the project config files nor version bumps to the Neo block plugin.

### Steps to reproduce

I think this is what I am asking: how can this be possible? A queue job mangling the data perhaps?

Otherwise:

Create an entry structure with two Neo blocks next two each other, add child blocks to each, do something to trigger that `“changes have been merged into this version”` message, then see if the child blocks were merged.

### Expected behaviour

The child blocks stay within their respective parent block lists.

### Neo version

4.0.7

### Craft CMS version

4.10.2

### Does this issue involve templating, and if so, is eager-loading used?

This is not a templating issue

**Comment by TheFunkyMonk** [comment]
I had a [similar issue](https://github.com/spicywebau/craft-neo/issues/937) (which just came up again) and running garbage collection with `php craft gc/run` fixed it; hopefully, that helps. Definitely something strange going on with garbage collection related to Neo.

**Comment by ttempleton** [comment]
Yes - try to run garbage collection as @TheFunkyMonk has suggested (after upgrading to the latest Neo version, currently 4.2.16) and let me know if it happens again.

**Comment by ttempleton** [comment]
Closing due to inactivity; if the issue persists with the latest version, please let me know, and we'll reopen it.
